tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,407 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Testing;
+using Aspire.Hosting.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.Oracle.Tests;
+
+public class OracleFunctionalTests
+{
+    private readonly ITestOutputHelper _testOutputHelper;
+
+    public OracleFunctionalTests(ITestOutputHelper testOutputHelper)
+    {
+        _testOutputHelper = testOutputHelper;
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyOracleResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() {
+                MaxRetryAttempts = int.MaxValue,
+                BackoffType = DelayBackoffType.Linear,
+                ShouldHandle = new PredicateBuilder().HandleResult(false),
+                Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var oracleDbName = "freepdb1";
+
+        var oracle = builder.AddOracle("oracle");
+        ConfigureTestOracleDatabase(oracle);
+
+        oracle.WithEnvironment("ORACLE_PASSWORD", oracle.Resource.PasswordParameter.Value);
+
+        var db = oracle.AddDatabase(oracleDbName);
+
+        var ready = builder.WaitForText(oracle.Resource, "Completed: ALTER DATABASE OPEN", cts.Token);
+
+        using var app = builder.Build();
+
+        await app.StartAsync(cts.Token);
+        await ready.WaitAsync(TimeSpan.FromMinutes(2));
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            using var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            return await databaseCreator.CanConnectAsync(token);
```

Comment:
Why aren't we doing some sort of SQL operation in this test? Like we were in the EndToEnd tests?

```
                var results = context.Database.SqlQueryRaw<int>("SELECT 1 FROM DUAL");
                return results.Any() ? Results.Ok("Success!") : Results.Problem("Failed");
```